### PR TITLE
Add continuous integration unit testing through GH Actions

### PR DESCRIPTION
**Pull request by jdebacker**
This PR added a GH Action to run unit tests on pushes.

I have a work in progress label because I can't get unit tests to run locally.  I just rebuilt the `toxbrain-dev` environment and when I run `pytest`, I get:

```
(taxbrain-dev) jason.debacker@JDEBACKER taxbrain % pytest
Traceback (most recent call last):
  File "/Users/jason.debacker/anaconda3/envs/taxbrain-dev/bin/pytest", line 7, in <module>
    from py.test import console_main
  File "/Users/jason.debacker/anaconda3/envs/taxbrain-dev/lib/python3.8/site-packages/py/_vendored_packages/apipkg/__init__.py", line 199, in __getattribute__
    return getattr(getmod(), name)
  File "/Users/jason.debacker/anaconda3/envs/taxbrain-dev/lib/python3.8/site-packages/py/_vendored_packages/apipkg/__init__.py", line 183, in getmod
    x = importobj(modpath, None)
  File "/Users/jason.debacker/anaconda3/envs/taxbrain-dev/lib/python3.8/site-packages/py/_vendored_packages/apipkg/__init__.py", line 72, in importobj
    module = __import__(modpath, None, None, ['__doc__'])
  File "/Users/jason.debacker/anaconda3/envs/taxbrain-dev/lib/python3.8/site-packages/pytest/__init__.py", line 5, in <module>
    from _pytest.assertion import register_assert_rewrite
  File "/Users/jason.debacker/anaconda3/envs/taxbrain-dev/lib/python3.8/site-packages/_pytest/assertion/__init__.py", line 9, in <module>
    from _pytest.assertion import rewrite
  File "/Users/jason.debacker/anaconda3/envs/taxbrain-dev/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line 34, in <module>
    from _pytest.assertion import util
  File "/Users/jason.debacker/anaconda3/envs/taxbrain-dev/lib/python3.8/site-packages/_pytest/assertion/util.py", line 13, in <module>
    import _pytest._code
  File "/Users/jason.debacker/anaconda3/envs/taxbrain-dev/lib/python3.8/site-packages/_pytest/_code/__init__.py", line 2, in <module>
    from .code import Code
  File "/Users/jason.debacker/anaconda3/envs/taxbrain-dev/lib/python3.8/site-packages/_pytest/_code/code.py", line 942, in <module>
    @attr.s(eq=False)
TypeError: attrs() got an unexpected keyword argument 'eq'
```

Once I am able to confirm tests run locally, I'll remove the WIP label.

### CODE REVIEW

**Comment by jdebacker** [comment]
Also, this PR would replace PR #22.  I think GH Actions is a better solution than Travis at this point.

**Comment by jdebacker** [comment]
Ok, solved error above which was due to a deprecated version of a package that was installed due to package conflicts.  Removing the `pdflatex` dependency, which was never imported in any module, solved this.

All tests (including Compute Studio app tests) are passing locally for me.  Looking into why failures on the GH Action.

**Comment by jdebacker** [comment]
Took a few tries to figure out how to properly install `texlive` properly on the virtual machine, but this PR now ready for your review @andersonfrailey.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ jobs:
         shell: bash -l {0}
         run: |
           sudo apt-get install texlive
+          sudo apt-get install texlive-latex-extra
```
Comment (hdoupe) [inline .github/workflows/build_and_test.yml:28]: @jdebacker You may be able to just do it in one go:

```
sudo apt-get install texlive texlive-latex-extra
```

**Comment by jdebacker** [comment]
@hdoupe Thanks for suggestion - one line worked!

**Comment by hdoupe** [comment]
> @hdoupe Thanks for suggestion - one line worked!

Awesome, thanks for stepping up and figuring out how to get this to work.

**Comment by jdebacker** [comment]
This PR addresses Issue #106

**Comment by andersonfrailey** [comment]
Thanks for working on this, @jdebacker!  LGTM. Merging.

**Pull request merged**